content-scope-scripts/content-scope-scripts-impl/src/main/java/com/duckduckgo/contentscopescripts/impl/messaging/ContentScopeScriptsJsMessaging.kt

```diff
@@ -54,31 +55,25 @@ class ContentScopeScriptsJsMessaging @Inject constructor(
     override val secret: String = coreContentScopeScripts.secret
     override val allowedDomains: List<String> = emptyList()
 
-    private val handlers: List<JsMessageHandler> = listOf()
+    private val handlers: List<JsMessageHandler> = listOf(WebShareHandler())
 
     @JavascriptInterface
     override fun process(message: String, secret: String) {
         try {
-            logcat(DEBUG) { "Message received $message" }
             val adapter = moshi.adapter(JsMessage::class.java)
             val jsMessage = adapter.fromJson(message)
             val domain = runBlocking(dispatcherProvider.main()) {
                 webView.url?.toUri()?.host
             }
             jsMessage?.let {
-                if (context == jsMessage.context && (allowedDomains.isEmpty() || allowedDomains.contains(domain))) {
+                if (this.secret == secret && context == jsMessage.context && (allowedDomains.isEmpty() || allowedDomains.contains(domain))) {
                     handlers.firstOrNull {
                         it.method == jsMessage.method && it.featureName == jsMessage.featureName
-                    }?.let {
-                        val response = it.process(jsMessage, secret, webView, jsMessageCallback)
-                        response?.let {
-                            jsMessageHelper.sendJsResponse(response, callbackName, secret, webView)
-                        }
-                    }
+                    }?.process(jsMessage, secret, webView, jsMessageCallback)
                 }
             }
         } catch (e: Exception) {
-            logcat { "Exception is ${e.message}" }
+            logcat { "Exception is ${e.stackTraceToString()}" }
```

Comment:
nit: I'd change it to `logcat(LogPriority.ERROR) { "Exception is ${e.asLog()} }`